common/TextDifferencer.java

```diff
@@ -16,16 +16,9 @@
 
 package com.google.startupos.common;
 
-import com.google.startupos.common.TextChange.Type;
-import java.util.ArrayList;
-import java.util.Collections;
 import java.util.List;
-import java.util.stream.Collectors;
-import java.util.stream.IntStream;
-import javax.swing.text.Segment;
 
-/** An implementation of text difference based on the Longest Common Subsequence (LCS) problem. */
-public class TextDifferencer {
+public interface TextDifferencer {
```

Comment:
If there's only one implementation (`LCSTextDifferencer`) then you don't need an interface, and then you also don't need `provideDefaultTextDifferencer()`.
If in the future we'll have 2 implementations, we can introduce an interface then.